app/controllers/participants_controller.rb

```diff
@@ -16,14 +18,21 @@ def new
   end
 
   def show
+    unless @participant.email_confirmed?
+      flash[:alert] = email_confirmation_alert(@participant)
+    end
```

Comment:
I thought about moving this to the applications controller (or layout), which might make sense, although I haven't seen an easy way to close or dismiss a flash message yet, so it might also be kind of annoying if it persists across all pages.

Reply:
Have you considered `flash.now[:alert]`?

Reply:
I think persisting is probably a-ok, since it does limit functionality. We'll massage the flash message and perhaps make it stickier as it shakes out.